pkg/controller/atlasproject/alert_configurations.go

```diff
@@ -48,54 +50,69 @@ func (r *AtlasProjectReconciler) ensureAlertConfigurations(service *workflow.Con
 }
 
 // This method reads secrets refs and fills the secret data for the related Notification
-func readAlertConfigurationsSecretsData(kubeClient client.Client, projectNs string, alertConfigs []mdbv1.AlertConfiguration) error {
+func (r *AtlasProjectReconciler) readAlertConfigurationsSecretsData(project *mdbv1.AtlasProject, alertConfigs []mdbv1.AlertConfiguration) error {
+	resourcesToWatch := make([]watch.WatchedObject, 0)
+	projectNs := project.Namespace
+	defer func() {
+		r.EnsureMultiplesResourcesAreWatched(types.NamespacedName{Namespace: project.Namespace, Name: project.Name}, r.Log, resourcesToWatch...)
+		r.Log.Debugf("watching alert configuration secrets: %v\r\n", r.WatchedResources)
+	}()
```

Comment:
Why is a defer function needed, rather than just putting this code after the for loop?

Reply:
Yes, the operator should watch for a specified secret even if there is no fields for a specific alert configuration. E.g: user configures one AtlasProject instance with an alert configuration using secret. The secret is missing the required data. If we won't add this secret to watcher, changes to this secret won't trigger reconciliation of a Project. I'll update the code to match that.

Reply:
ok